pallets/transaction-multi-payment/src/lib.rs

```diff
@@ -371,15 +371,17 @@ impl<T: Config> Pallet<T> {
 		Authorities::<T>::mutate(|x| x.push(who.clone()));
 	}
 
-	pub fn withdraw_set_fee(who: &T::AccountId, currency: AssetId) -> DispatchResult {
+	pub fn withdraw_set_fee(who: &T::AccountId) -> DispatchResult {
 		let base_fee = Self::weight_to_fee(T::BlockWeights::get().get(DispatchClass::Normal).base_extrinsic);
 		let adjusted_weight_fee = Self::weight_to_fee(T::WeightInfo::set_currency());
 		let fee = base_fee.saturating_add(adjusted_weight_fee);
 
-		Self::swap_currency(who, fee)?;
-		T::MultiCurrency::withdraw(currency, who, fee)?;
-
-		Ok(())
+        let result = Self::swap(who, fee)?;
+        match result {
+			PaymentSwapResult::Transferred => Ok(()),
+			PaymentSwapResult::Native | PaymentSwapResult::Swapped => T::MultiCurrency::withdraw(CORE_ASSET_ID, who, fee),
```

Comment:
We need to check the result of withdraw too. Because if it fails, the tx should fail too.

Reply:
I believe that the result (`withdraw` method returns `DispatchResult`) is returned from the `match` statement. This test should prove it https://github.com/galacticcouncil/Basilisk-node/blob/076984b3988833b1144b5a397520b3fad21f0b9a/pallets/transaction-multi-payment/src/tests.rs#L142
Or am I wrong?

Reply:
yes. correct.  I haven;t realized it.